lib/search.coffee

```diff
@@ -72,9 +73,14 @@ createSearch = ({neighborhood})->
                 return "{{#{p}}}"
               else return p
             .join('')
-          text: result.page.synopsis.split(new RegExp(searchQuery, 'i')).join("{{#{searchQuery}}}") || '',
+          text: result.page.synopsis
+            .split(new RegExp("(#{searchQuery})", 'i'))
+            .map (p) ->
+              if searchQuery.toLowerCase() == p.toLowerCase()
+                return "{{#{p}}}"
+              else return p
+            .join('')
```

Comment:
I'm trying to follow along at home. Can you remind me what {{ }} does in this code? How does this come to have the undesirable effect you corrected?

Reply:
`{{ }}` are markers that surround the search term so I can find it after it has gone through rendering by the `reference` plugin (without the risk of breaking the markup generated by the `reference` plugin). This particular `if` clause was added to preserve the case of the search term as it appears in the original source rather than using the case of the search term as typed by the user.

The current version of this code does the following:
* Perform a capturing case insensitive match of the search term.
* Surround all instances of the match with the `{{ }}` markers while preserving the casing of the search term from the original source.
* Render the search matches through the `reference` plugin.
* Find all instances of the now rendered search terms and surround them with a `<span/>` to add highlighting.

Reply:
Ah. I knew there was something to it that I didn't yet appreciate. Thanks for explaining.